audapter_viewer/audapter_viewer.m

```diff
@@ -198,6 +198,31 @@
 % For recalculating everything 
 bAllOstRecalculated = 1; 
 bTheseOstRecalculated = 1; 
+% Initialize calcSubjOstParams if it doesn't already exist
+if ~isfield(y, 'calcSubjOstParams')
+    % Some backwards compatibility if you are opening up a study that didn't store parameters by trial 
+    if strcmp(methodUsed, 'data.subjOstParams')
+        [y.calcSubjOstParams] = y(:).subjOstParams; 
+
+    elseif strcmp(methodUsed, 'expt.calcSubjOstParams')
+        for i = 1:length(y)
+            y(i).calcSubjOstParams = x.calcSubjOstParams; 
+        end
+
+    elseif strcmp(methodUsed, 'expt.subjOstParams')
+        for i = 1:length(y)
+            y(i).calcSubjOstParams = x.subjOstParams; 
+        end
+
+    elseif strcmp(methodUsed, 'refreshWorkingCopy')
+        warning('No OST params saved for this participant (may be an old study, or not have used OSTs). calcSubjOstParams not initiated.');         
+        subjOstParams = get_ost(trackingFileDir, trackingFileName, 'full', 'master'); 
+        for i = 1:length(y)
+            y(i).calcSubjOstParams = subjOstParams; 
+        end
+            
+    end
+end
```

Comment:
Good catch on 'none'! For the other example, if data.calcSubjOstParams is the method, then you wouldn't have to initiate the field because it already exists (y is data in audapter_viewer)---it'd be caught by the outer if check.